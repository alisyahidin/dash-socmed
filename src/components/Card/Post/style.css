.post {
  padding: 1rem 0rem;
  border-bottom: 1px solid rgba(196, 222, 255, 0.5);
}

.post:last-child {
  border-bottom: none;
}

.post__author:hover {
  text-decoration: none;
}

.post__body:hover {
  cursor: pointer;
}

.author__username {
  color: #888;
}

.author__username:hover {
  color: #666;
}