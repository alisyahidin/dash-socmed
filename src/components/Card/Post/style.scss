.post {
  padding: 1rem 0rem;
  border-bottom: 1px solid rgba(196, 222, 255, 0.5);
  &:last-child {
    border-bottom: none;
  }
  &__title,
  &__body {
    transition: all 300ms ease;
    &:hover {
      cursor: pointer;
      color: #555;
    }
  }
}

.author {
  &:hover {
    text-decoration: none;
  }
  &__username {
    transform: scale(0.8);
    color: #888;
  }
  &__username:hover {
    color: #666;
  }
}