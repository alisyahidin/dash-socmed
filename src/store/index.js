import { createStore, applyMiddleware } from 'redux'
import { createEpicMiddleware } from 'redux-observable'
import { from } from 'rxjs'
import axios from '../lib/api'

import rootReducer from '../reducers/'
import { rootEpic } from '../epics/'

const epicMiddleware = createEpicMiddleware({
  dependencies: {
    axios$: (url, data = null) =>
      from(axios(url, data))
  }
})
const store = createStore(rootReducer, applyMiddleware(epicMiddleware))
epicMiddleware.run(rootEpic)

export default store