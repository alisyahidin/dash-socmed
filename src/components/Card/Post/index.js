import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Link } from 'react-router-dom'
import { Row, Col } from 'react-bootstrap'

import { openModal } from '../../../actions/singlePost'
import './style.scss'

const Post = ({author, id, title, body, openModal, ...props}) => {
  return (
    <>
      <Row className="post mx-sm-1">
        <Col className="p-sm-0 d-flex flex-column justify-content-center align-items-start">
          <h5 onClick={() => openModal(true, id)} className="post__title mb-sm-2">{title}</h5>
          <span>
            by&nbsp;-&nbsp;
            <Link to={`/users/${author.id}`} className="author d-inline-block mb-sm-1">
              <b className="mr-sm-1">
                {author.name}
              </b>
              <span className="author__username mr-sm-2">
                @{author.username}
              </span>
            </Link>
          </span>
          <div onClick={() => openModal(true, id)} className="post__body">
            <p className="mb-sm-2">{body}</p>
          </div>
        </Col>
      </Row>
    </>
  )
}

Post.propTypes = {
  id: PropTypes.number.isRequired,
  author: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
}

Post.defaultProps = {
  id: 0,
  author: {
    id: 1,
    name: 'Guest',
    username: 'guest'
  },
  title: 'Title',
  body: 'Post',
}

const mapStateToProps = state => ({ ...state })

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    openModal
  }, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(Post)
