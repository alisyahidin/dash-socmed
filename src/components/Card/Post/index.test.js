import React from 'react'
import { MemoryRouter } from 'react-router'
import { Provider } from 'react-redux'
import { shallow, mount } from 'enzyme'
import ReactRouterEnzymeContext from 'react-router-enzyme-context'
import store from '../../../store/'
import Post from './index'

it('render <Post /> correctly', () => {
  const component = shallow(<Post />)

  expect(component).toMatchSnapshot()
})