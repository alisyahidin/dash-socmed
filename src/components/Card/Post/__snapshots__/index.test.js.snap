// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Post /> correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
